packages/freezed/test/multiple_constructors_test.dart

```diff
@@ -289,4 +279,57 @@ void main() {
       error,
     );
   });
+  group('NestedList', () {
+    test('generates List of correct type', () async {
+      final nestedListClass = await _getClassElement('ShallowNestedList');
+
+      expect(nestedListClass.getField('children').type.getDisplayString(),
+          'List<LeafNestedListItem>');
+    });
+
+    test('generates List of correct type for deeply nested case', () async {
+      final nestedListClass = await _getClassElement('DeepNestedList');
+
+      expect(nestedListClass.getField('children').type.getDisplayString(),
+          'List<InnerNestedListItem>');
+
+      final nestedListItemClass = await _getClassElement('InnerNestedListItem');
+
+      expect(nestedListItemClass.getField('children').type.getDisplayString(),
+          'List<LeafNestedListItem>');
+    });
+  });
+  group('NestedMap', () {
+    test('generates Map of correct type', () async {
+      final nestedMapClass = await _getClassElement('ShallowNestedMap');
+
+      expect(nestedMapClass.getField('children').type.getDisplayString(),
+          'Map<String, LeafNestedMapItem>');
+    });
+
+    test('generates Map of correct type for deeply nested case', () async {
+      final nestedMapClass = await _getClassElement('DeepNestedMap');
```

Comment:
_getClassElement is very expensive. It shouldn't be called more than once per test.